### PR TITLE
Do more stuff from main macros

### PR DESCRIPTION
**Pull request by keszybz**
Replacement for #10851, #10849, #10848.

### CODE REVIEW

**Comment by poettering** [comment]
label bingo, eh?

**Comment by poettering** [comment]
(i think we should update the label colors so that when they are all turned on they are a pretty rainbow ;-))

File: src/cgls/cgls.c [hunk 1]
```diff
@@ -36,6 +37,8 @@ static char **arg_names = NULL;
 static int arg_full = -1;
 static const char* arg_machine = NULL;
 
+STATIC_DESTRUCTOR_REGISTER(arg_names, freep);
```
Comment (poettering) [inline src/cgls/cgls.c]: should retain the comment the free() call mentioned before: `/* don't free the strings */`

**Comment by poettering** [comment]
looks great to me, but would like the one comment to remain in there, see above.

**Comment by yuwata** [comment]
What a huge number of labels...

File: src/cgls/cgls.c (hunk 1 above)
Comment (keszybz) [reply src/cgls/cgls.c]: Updated.

**Comment by poettering** [comment]
CI fails, and afaics rightfully

File: src/systemctl/systemctl.c [hunk 2]
```diff
@@ -8749,6 +8746,8 @@ int main(int argc, char*argv[]) {
         free(arg_root);
         free(arg_esp_path);
 
-        /* Note that we return r here, not EXIT_SUCCESS, so that we can implement the LSB-like return codes */
-        return r < 0 ? EXIT_FAILURE : r;
+        /* Note that we return r here, not 0, so that we can implement the LSB-like return codes */
+        return r;
 }
+
+DEFINE_MAIN_FUNCTION(run);
```
Comment (yuwata) [inline src/systemctl/systemctl.c]: This should be `DEFINE_MAIN_FUNCTION_WITH_POSITIVE_FAILURE`.

**Comment by keszybz** [comment]
Updated.

**Pull request merged**